### DISCUSSION TITLE
Scale input/sensor value before send MQTT

### DISCUSSION THREAD

**Discussion by Corrado-Cuneo** [arendst/Tasmota#12675] [Q&A] [answered]
Hello!!! I've a NodeMCU with Tasmota with a distance sensore SR-04.... work fine!!

I'use it to measure a tank water level, where 40cm >> Empty, 3cm Full...

I need to convert the sensore value as this:

40 cm --> 0 litre

3cm --> 25 litre

What function can I use? Scale?

Thanks!!

Corrado

**Comment by joba-1** [comment, marked as answer]
assuming your top surface does not change with the level you solve a linear equation and get

liter = -25/37 level + 1000/37

**Reply by sfromis** [reply]
To get the value converted, try a rule with `Scale1 %value%,3,40,25,3`

**Comment by Corrado-Cuneo** [comment]
Thanks!!! Have you an example of role creation with Scale function?
Corrado

**Reply by sfromis** [reply]
`Rule1 on SR04#Distance do Scale1 %value%,3,40,25,3 endon`
Remember to activate the rule. After that rule, the scaled value is available in var1, available to other rules.

**Reply by Corrado-Cuneo** [reply]
Thanks!!!
and how can I pubblish var1 via MQTT after scale?
This evening I try it!!

**Reply by sfromis** [reply]
Take the learning curve, see https://tasmota.github.io/docs/Rules/
